field.cpp

```diff
@@ -0,0 +1,194 @@
+#include "field.h"
+#include <random>
+#include <utility>
+
+Field::Field(int fieldSize){
+    setSize(fieldSize);
+    isMovable = true;
+}
+
+void Field::setSize(int fieldSize){
```

Comment:
A minor but important thing, all the spaces have to be the same so here you have ){ and below near the for loop you have ) {
It is better to use ) { in this case everywhere

Reply:
done